src/error.c

```diff
@@ -152,22 +152,44 @@ Obj FuncCURRENT_STATEMENT_LOCATION(Obj self, Obj context)
     return retlist;
 }
 
-Obj FuncPRINT_CURRENT_STATEMENT(Obj self, Obj context)
+
+Obj FuncPRINT_CURRENT_STATEMENT(Obj self, Obj stream, Obj context)
 {
     if (context == STATE(BottomLVars))
         return 0;
 
+    /* HACK: we want to redirect output */
+    /* Try to print the output to stream. Use *errout* as a fallback. */
+    if ((IsStringConv(stream) && !OpenOutput(CSTR_STRING(stream))) ||
+        (!IS_STRING(stream) && !OpenOutputStream(stream))) {
+        if (OpenOutput("*errout*")) {
+            Pr("PRINT_CURRENT_STATEMENT: can not open error stream\n", 0, 0);
+        }
+        else {
+            int ret = fputs("gap: panic, can not open *errout*!\n", stderr);
+            /* If that failed, try printing to stdout */
```

Comment:
Hmm, why do this?  Is there a specific problem case you have in mind there? Why would printing to stderr fail? And if it fails,  would printing to stdout still work? And, are we really that desperate to print this message?

Reply:
I didn't have any specific situation in mind. I thought a fallback option for when *errout* can't be openend would make sense. As to why first try stderr and then stdout: I saw similar code in `PRINT_OR_APPEND_TO` in `streams.c` and thought "better be safe than sorry".
Should I change this to just print to stderr and exit directly?

Reply:
I changed this to be consistent with basically any other "GAP panic".